src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -982,13 +983,14 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof SSLCallback) {
                     SSLCallback sslCallback = (SSLCallback) callback;
-
+                    SSLSession sslSession = sslCallback.getSslSession();
                     try {
-                        X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslCallback.getSslSession().getPeerCertificates());
+                        X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslSession.getPeerCertificates());
                         verifyEvidence(new X509PeerCertificateChainEvidence(x509Certificates));
                         certs = x509Certificates;
                     } catch (SSLPeerUnverifiedException e) {
-                        log.trace("Peer unverified", e);
+                        log.trace("Peer unverified. Using server certificates instead", e);
+                        certs = X500.asX509CertificateArray(sslSession.getLocalCertificates());
```

Comment:
I should clarify; the peer certificate should still be verified so that EXTERNAL works.  But the `certs` field should always be populated from the local certificates.